src/webui/api/torrentscontroller.cpp

```diff
@@ -110,20 +115,94 @@ namespace
     using Utils::String::parseBool;
     using Utils::String::parseTriStateBool;
 
-    void applyToTorrents(const QStringList &hashes, const std::function<void (BitTorrent::TorrentHandle *torrent)> &func)
+    void applyToTorrents(const QStringList &hashes, const std::function<void (BitTorrent::TorrentHandle *const torrent)> &func)
```

Comment:
Please revert this change (and all the related changes). It's not required by your feature and it's questionable (absolutely meaningless for me personally) to apply it in passing.

Reply:
I don't see any reason we wouldn't want this change. In the future I'll move unrelated changes to their own PR; this commit is already a part of this so no point in moving now.

Reply:
>I don't see any reason we wouldn't want this change. 

The reason is at least the changes in this line makes functor declaration to be confusing (having "const" applied to parameter passed by value in function declaration is meaningless and confusing).
Having "const" in the following places of this function has minimal meaning since the function code is too trivial and it isn't supposed to become any more complex in the future (but there are no errors formally and some adherents of the "paranoid" safety can insist on its use). The same can be said about the definition of methods that we pass in "applyToTorrents()" when calling (usually it is some one-liner).
Well, I'm not going to insist on the revert of the formally correct changes, if others deem them acceptable. But I insist on never having "const" applied to "byval" parameters in function declaration.

Reply:
> But I insist on never having "const" applied to "byval" parameters in function declaration.

On this point we agree

> but there are no errors formally and some adherents of the "paranoid" safety can insist on its use

That's mainly my goal here. The point *can* be const, so why not enforce it.

Reply:
Ok, fix this one and let's proceed.

Reply:
Sorry, I don't follow. Is your issue with this line `BitTorrent::TorrentHandle *const torrent`?

Reply:
Yes.
"std::function" template argument is similar to function declaration so it should be affected by the restriction above.

Reply:
But we aren't passing the torrent by value, we're passing a pointer to it.

Reply:
>we're passing a pointer to it

Yes. We're passing "pointer to torrent" by value.

Reply:
Ah I see. What would the syntax be to have the pointer passed by pointer/reference?

Reply:
Pointer by reference: `TorrentHandle *&torrent`.
"Pointer by pointer" is incorrect term. You can just have "pointer to pointer" (or even "pointer to pointer to pointer ... to pointer") and pass it either by value or by reference.